src/sql/workbench/browser/designer/interfaces.ts

```diff
@@ -129,7 +129,7 @@ export interface DesignerDataPropertyInfo {
 	componentProperties?: InputBoxProperties | CheckBoxProperties | DropDownProperties | DesignerTableProperties;
 }
 
-export type DesignerComponentTypeName = 'input' | 'checkbox' | 'dropdown' | 'table';
+export type DesignerComponentTypeName = 'input' | 'checkbox' | 'dropdown' | 'table' | 'editableDropdown';
```

Comment:
hmm, I'd prefer not to introduce a new component type, but add an option `editable` to the existing dropdown component properties. also, don't forget to update the typings file.